components/switch/index.css

```
@@ -466,28 +461,27 @@
 				&:not(:checked) + .spectrum-Switch-switch {
 					box-shadow: inset 0 0 0 1px GrayText;
 					background-color: ButtonFace;
+				}
 
-					&::before {
-						border-color: GrayText;
-						background-color: ButtonFace;
-					}
+				&:not(:checked) + .spectrum-Switch-switch::before {
+					border-color: GrayText;
+					background-color: ButtonFace;
 				}
 
 				&:checked + .spectrum-Switch-switch {
 					box-shadow: inset 0 0 0 1px GrayText;
 					background-color: GrayText;
+				}
 
-					&::before {
-						border-color: GrayText;
-						background-color: ButtonFace;
-					}
+				&:checked + .spectrum-Switch-switch	::before {
```

```suggestion
                                &:checked + .spectrum-Switch-switch::before {
```
just a space deletion we could make here. 👍